prow/cmd/deck/job_history.go

```diff
@@ -353,18 +428,18 @@ func (a int64slice) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 func (a int64slice) Less(i, j int) bool { return a[i] < a[j] }
 
 // Gets job history from the GCS bucket specified in config.
-func getJobHistory(url *url.URL, config *config.Config, gcsClient *storage.Client) (jobHistoryTemplate, error) {
+func getJobHistory(ctx context.Context, url *url.URL, config *config.Config, gcsClient *storage.Client, opener io.Opener) (jobHistoryTemplate, error) {
 	start := time.Now()
 	tmpl := jobHistoryTemplate{}
 
-	bucketName, root, top, err := parseJobHistURL(url)
+	storageType, bucketName, root, top, err := parseJobHistURL(url)
 	if err != nil {
 		return tmpl, fmt.Errorf("invalid url %s: %v", url.String(), err)
 	}
 	tmpl.Name = root
-	bucket := gcsBucket{bucketName, gcsClient.Bucket(bucketName)}
+	bucket := gcsBucket{bucketName, storageType, gcsClient.Bucket(bucketName), opener}
```

Comment:
Could you make a commit at the end where you rename `gcsBucket` to something more neutral like `objStorageBucket`?

Reply:
I added it to the PR description.